source/server/admin/stats_params.cc

```diff
@@ -0,0 +1,42 @@
+#include "source/server/admin/stats_params.h"
+
+namespace Envoy {
+namespace Server {
+
+Http::Code StatsParams::parse(absl::string_view url, Buffer::Instance& response) {
+  query_ = Http::Utility::parseAndDecodeQueryString(url);
+  used_only_ = query_.find("usedonly") != query_.end();
+  pretty_ = query_.find("pretty") != query_.end();
+  prometheus_text_readouts_ = query_.find("text_readouts") != query_.end();
+  if (!Utility::filterParam(query_, response, filter_)) {
+    return Http::Code::BadRequest;
```

Comment:
Might be outside the scope of this change, but it would be helpful if the response body said which param made it a bad request

Reply:
filterParam() has in it:
```
      response.add(fmt::format("Invalid regex: \"{}\"\n", error.what()));
```
is that sufficient?